dash-spv/src/storage/disk/segments.rs

```
@@ -1,322 +1,695 @@
+            let key_to_evict =
+                segments.iter_mut().min_by_key(|(_, s)| s.last_accessed).map(|(k, v)| (*k, v));
 
-/// Evict the oldest (least recently accessed) segment.
-pub(super) async fn evict_oldest_segment(
-    manager: &DiskStorageManager,
-    segments: &mut HashMap<u32, SegmentCache>,
-) -> StorageResult<()> {
-    if let Some(oldest_id) = segments.iter().min_by_key(|(_, s)| s.last_accessed).map(|(id, _)| *id)
-    {
-        // Get the segment to check if it needs saving
-        if let Some(oldest_segment) = segments.get(&oldest_id) {
-            // Save if dirty or saving before evicting - do it synchronously to ensure data consistency
-            if oldest_segment.state != SegmentState::Clean {
-                tracing::debug!(
-                    "Synchronously saving segment {} before eviction (state: {:?})",
-                    oldest_segment.segment_id,
-                    oldest_segment.state
-                );
-                let segment_path = manager
-                    .base_path
-                    .join(format!("headers/segment_{:04}.dat", oldest_segment.segment_id));
-                super::io::save_segment_to_disk(&segment_path, &oldest_segment.headers).await?;
-                tracing::debug!("Successfully saved segment {} to disk", oldest_segment.segment_id);
+            if let Some((key, segment)) = key_to_evict {
+                segment.persist(&self.base_path)?;
+                segments.remove(&key);
             }
         }
 
-        segments.remove(&oldest_id);
+        // Load and insert
+        let segment = Segment::load(&self.base_path, *segment_id).await?;
+        let segment = segments.entry(*segment_id).or_insert(segment);
+        Ok(segment)
     }
 
-    Ok(())
-}
+    pub async fn get_items(&mut self, height_range: Range<u32>) -> StorageResult<Vec<I>> {
+        if height_range.start > height_range.end {
+            panic!("Invalid height range: start > end, {:?}", height_range);
```

Shouldn't you rather return an error here?